src/events/crawler/lib/fetch.js

```diff
@@ -147,30 +147,49 @@ const fetchHeadless = async url => {
   await page.setUserAgent(CHROME_AGENT);
   await page.setViewport(DEFAULT_VIEWPORT);
 
-  try {
-    const response = await page.goto(url, {
-      timeout: READ_TIMEOUT,
-      waitUntil: 'networkidle2'
-    });
+  let tries = 0;
+  while (tries < 5) {
+    tries++;
+    if (tries > 1) {
+      // sleep a moment before retrying
+      console.log(`  ⚠️  retrying (${tries})`);
+      await new Promise(r => setTimeout(r, 2000));
+    }
+
+    try {
+      const response = await page.goto(url, {
+        timeout: READ_TIMEOUT,
+        waitUntil: 'networkidle2'
+      });
 
-    if (response._status < 400) {
-      await page.waitFor(RESPONSE_TIMEOUT);
-      const html = await page.content();
+      if (response === null) {
+        browser.close();
+        continue;
+      }
+
+      if (response._status < 400) {
+        await page.waitFor(RESPONSE_TIMEOUT);
+        const html = await page.content();
+        browser.close();
+        return html;
+      }
+      if (response._status < 500) {
+        console.log(`  ❌ Got error ${response._status} trying to fetch ${url}`);
+        browser.close();
+        return null;
+      }
```

Comment:
Will the server ever have an internal server error (code >= 500)?  And if so, should it retry?  E.g., see the checks in https://github.com/lazd/coronadatascraper/blob/master/src/events/crawler/lib/get.js#L80.

Reply:
(I'm assuming that a >=500 error wouldn't actually result in an error being thrown)

Reply:
Hmm, I think retrying on an internal server error is likely to just result in another failure (trying 30 minutes later, maybe not), but it's possible we could get it right as it's fixed...

Reply:
Servers return 503 or other 5xx all the time when they are simply overloaded. Sleeping a bit and retrying is a perfectly great thing to do.